internal/collector/recon/sharding.go

```diff
@@ -0,0 +1,608 @@
+	Failure   int `json:"failure"`
+	Success   int `json:"success"`
+}
+
+type Misplaced struct {
+	Attempted int `json:"attempted"`
+	Failure   int `json:"failure"`
+	Found     int `json:"found"`
+	Placed    int `json:"placed"`
+	Success   int `json:"success"`
+	Unplaced  int `json:"unplaced"`
+}
+
+type Scanned struct {
+	Attempted int `json:"attempted"`
+	Failure   int `json:"failure"`
+	MaxTime   int `json:"max_time"`
+	MinTime   int `json:"min_time"`
+	Success   int `json:"success"`
+}
+
+type Visited struct {
+	Attempted int `json:"attempted"`
+	Completed int `json:"completed"`
+	Failure   int `json:"failure"`
+	Skipped   int `json:"skipped"`
+	Success   int `json:"success"`
+}
+
+type ShardingInProgress struct {
+	All []struct {
+		Account     string `json:"account"`
+		Active      int    `json:"active"`
+		Cleaved     int    `json:"cleaved"`
+		Container   string `json:"container"`
+		Created     int    `json:"created"`
+		DBState     string `json:"db_state"`
+		Error       string `json:"error"`
+		FileSize    int64  `json:"file_size"`
+		Found       int    `json:"found"`
+		NodeIndex   int    `json:"node_index"`
+		ObjectCount int64  `json:"object_count"`
+		Path        string `json:"path"`
+		State       string `json:"state"`
+		Root        string `json:"root"`
+	} `json:"all"`
+}
+
+type Top struct {
+	Account     string `json:"account"`
+	Container   string `json:"container"`
+	FileSize    int64  `json:"file_size"`
+	NodeIndex   int    `json:"node_index"`
+	ObjectCount int64  `json:"object_count"`
+	Path        string `json:"path"`
+	Root        string `json:"root"`
+}
+
+type ShardingCandidates struct {
+	Found int   `json:"found"`
+	Top   []Top `json:"top"`
+}
+
+type Sharding struct {
+	AuditRoot          AuditRoot          `json:"audit_root"`
+	AuditShard         AuditShard         `json:"audit_shard"`
+	Cleaved            ShardCleaved       `json:"cleaved"`
+	Created            ShardCreated       `json:"created"`
+	Misplaced          Misplaced          `json:"misplaced"`
+	Scanned            Scanned            `json:"scanned"`
+	Visited            Visited            `json:"visited"`
+	ShardingInProgress ShardingInProgress `json:"sharding_in_progress"`
+	ShardingCandidates ShardingCandidates `json:"sharding_candidates"`
+}
+
+type ShardingStats struct {
+	Sharding Sharding `json:"sharding"`
+}
+
+// UpdateMetrics implements the collector.Task interface.
+func (t *ShardingTask) UpdateMetrics() (map[string]int, error) {
+	cmdArgs := t.cmdArgs
+	q := util.CmdArgsToStr(cmdArgs)
+	queries := map[string]int{q: 0}
+	e := &collector.TaskError{
+		Cmd:     "swift-recon",
+		CmdArgs: cmdArgs,
+	}
+
+	outputPerHost, err := getSwiftReconOutputPerHost(t.opts.CtxTimeout, t.opts.PathToExecutable, cmdArgs...)
+	if err != nil {
+		queries[q] = 1
+		e.Inner = err
+		return queries, e
+	}
+
+	for hostname, dataBytes := range outputPerHost {
+		if !json.Valid(dataBytes) {
+			dataBytes = bytes.ReplaceAll(dataBytes, []byte(`"None"`), []byte(`0`))
+		}
```

Comment:
It's not clear what this is about. Please add a code comment explaining the purpose of this section.

Reply:
Done

Reply:
Thanks for the clarification. Since this is an issue that can affect the entire Python literal -> JSON parsing, I pushed a commit that moves this to the central location. Also, I changed it to not rewrite into an unrelated string for clarity. Please take a look.

Reply:
Yes, this makes more sense. I was (incorrectly) operating under the assumption that there was a reason for it existing as it did, hence the localised workaround.